core/src/software/aws/toolkits/core/utils/RemoteResourceResolver.kt

```diff
@@ -61,7 +65,12 @@ class DefaultRemoteResourceResolver(
         when {
             downloaded != null -> {
                 LOG.debug { "Downloaded new file $downloaded, replacing old file $expectedLocation" }
-                Files.move(downloaded, expectedLocation, StandardCopyOption.REPLACE_EXISTING)
+                val fileStream = FileInputStream(downloaded.toFile())
+                val isParsingSuccess = resource.remoteResolveParser?.canBeParsed(fileStream) ?: true
+                fileStream.close()
```

Comment:
you can actually write it like this since it's closable:

```suggestion
                val isParsingSuccess = FileInputStream(downloaded.toFile()).use {
                    resource.remoteResolveParser?.canBeParsed(fileStream) ?: true
                }
```